src/tool/hpcrun/gpu/nvidia/cubin-symbols.c

```diff
@@ -280,12 +287,14 @@ computeCubinFunctionOffsets
   Elf_SymbolVector *symbols = NULL;
   elf_version(EV_CURRENT);
   Elf *elf = elf_memory(cubin_ptr, cubin_len);
+  elf_helper_t eh;
```

Comment:
@jmellorcrummey I find that there is a good reason to keep the `elf_helper_t` definition in the header file. If we move the definition to the .c file, I will need an interface allocation function to return a pointer to `elf_helper_t`. Here we are in hpcrun. So we don't want to use malloc if possible in the allocation function. And since the code will be used by hpcstruct, we cannot use hpcrun_malloc in the allocation function. 

The strategy I am using is to put the type definition in the header file, declare it on stack, passed the pointer of the stack variable around.

Reply:
good point. I agree.